src/Symfony/Component/Security/CHANGELOG.md

```diff
@@ -4,6 +4,8 @@ CHANGELOG
 5.3
 ---
 
+ * Deprecate class `User`, use `InMemoryUser` instead
```

Comment:
I think we should be more explicit that this is not a drop-in replacement, as methods related to advanced user functionality have been removed

Reply:
Done in UPGRADE files